packages/server/src/ws-connection.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  isResponseDataState,
+  isServerEventDataState,
+  lookupValue,
+  response,
+} from '@rise-tools/react'
+import { WebSocket } from 'ws'
+import { z } from 'zod'
+
+import { findModel, getModelState } from './model-utils'
+import { AnyModels, ValueModel } from './types'
+
+const serverSubscribeMessageSchema = z.object({
+  $: z.literal('sub'),
+  keys: z.array(z.string()),
+})
+const serverUnsubscribeMessageSchema = z.object({
+  $: z.literal('unsub'),
+  keys: z.array(z.string()),
+})
+const serverEventMessageSchema = z.object({
+  $: z.literal('evt'),
+  event: z.object({
+    target: z.object({
+      key: z.string().optional(),
+      component: z.string(),
+      propKey: z.string(),
+      path: z.array(z.string().or(z.number())),
```

Comment:
nit: 
```
z.tuple([z.string()]).rest(z.string().or(z.number()))
```
will give
```
type Path = [string, ...(string | number)[]] 
```
which will let you remove some `if` conditions, typically:
```
if (!storeName) throw new Error('Missing store name in event path')
if (typeof storeName !== 'string') throw new Error('Store name must be a string')
```

Reply:
lets skip this for now and fix it when we separate the model path from value path